pkg/scheduler/cache/cache.go

```diff
@@ -240,8 +242,112 @@ func (cache *schedulerCache) addPod(pod *v1.Pod) {
 	n.AddPod(pod)
 }
 
+// this function expects valid pod, and valid, non-empty resizeRequestAnnotation json string
+func (cache *schedulerCache) getPodResizeRequirements(pod *v1.Pod, resizeRequestAnnotation string) (map[string]v1.Container, *Resource, error) {
+	var resizeContainers []v1.Container
+
+	if err := json.Unmarshal([]byte(resizeRequestAnnotation), &resizeContainers); err != nil {
+		return nil, nil, err
+	}
+
+	resizeContainersMap := make(map[string]v1.Container)
+	for _, resizeContainer := range resizeContainers {
+		resizeContainersMap[resizeContainer.Name] = resizeContainer
+	}
+
+	podResource := &Resource{}
+	for _, container := range pod.Spec.Containers {
+		containerResourcesRequests := v1.ResourceList {
+							v1.ResourceCPU:    container.Resources.Requests[v1.ResourceCPU],
```

Comment:
consider doing containerResourcesRequests := container.Resources.Request.DeepCopy() instead in case later we need to deal with more resources than cpu and memory.

Reply:
done.